src/neo/Network/P2P/Payloads/OracleResponse.cs

```diff
@@ -0,0 +1,74 @@
+using Neo.IO;
+using Neo.IO.Json;
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.SmartContract.Native;
+using Neo.SmartContract.Native.Oracle;
+using Neo.VM;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class OracleResponse : TransactionAttribute
+    {
+        private const int MaxResultSize = 1024;
+
+        public static readonly byte[] FixedScript;
+
+        public ulong Id;
+        public bool Success;
+        public byte[] Result;
+
+        public override TransactionAttributeType Type => TransactionAttributeType.OracleResponse;
+        public override bool AllowMultiple => false;
+
+        public override int Size => base.Size +
+            sizeof(ulong) +         //Id
+            sizeof(bool) +          //Success
+            Result.GetVarSize();    //Result
+
+        static OracleResponse()
+        {
+            using ScriptBuilder sb = new ScriptBuilder();
+            sb.EmitAppCall(NativeContract.Oracle.Hash, "finish");
+            FixedScript = sb.ToArray();
+        }
+
+        protected override void DeserializeWithoutType(BinaryReader reader)
+        {
+            Id = reader.ReadUInt64();
+            Success = reader.ReadBoolean();
+            Result = reader.ReadVarBytes(MaxResultSize);
```

Comment:
We can serialize `Result` only if `Success`

Reply:
If not `Success`, the length is `0`.

Reply:
We can save the prefix length

Reply:
I changed it to an `OracleRepsonseCode`. We can't save the prefix now... 😅